### ISSUE TITLE
Selection using mouse clicks don't work

### ISSUE THREAD

**Issue by danger-ahead** [knadh/autocomp.js#1]
Currently selections with only keyboard seem to work, will it be possible to add mouse click as well?

**Comment by knadh** [comment]
Hi @danger-ahead. Clicking on the search results work: https://github.com/knadh/autocomp.js/blob/1985ae990fad019e6d991db65206217c94ac4a62/autocomp.js#L105

Could you share steps to reproduce what you're experiencing? Is this a mobile device?

**Comment by jxxt** [comment]
I encountered the same issue.

I am testing on this [website](https://knadh.github.io/autocomp.js/):
Suppose I type "a" in the input box, and if I click "apple" using my mouse....it doesn't gets selected and the title id h2 element remains as it is. But it works when I press enter on my keyboard to select an item.

device: windows 11
browser: Google Chrome, version 116.0.5845.111

**Comment by AndrewBastin** [comment]
@knadh a very easy way to reproduce this behaviour is just to click and hold the result instead of just normally clicking it. 

This seems to be a small race condition that usually happens with how browsers handle the `click` and `blur` events on the clicked auto complete item box and the (now unfocused) input element.

It seems that this case is handled by using a `setTimeout` to queue the removal of the box:
https://github.com/knadh/autocomp.js/blob/1985ae990fad019e6d991db65206217c94ac4a62/autocomp.js#L20-L24

But the 100ms delay prescribed here is not exactly a good fix as this depends on when the browser deems the click is 'complete'. The `blur` event due to the focus being lost will be fired as soon as the mouse is down (and hence, the focus shifts to the clicked item), but the `click` event will not fire until the mouse button is released ([relevant MDN quote](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#:~:text=An%20element%20receives%20a%20click%20event%20when%20a%20pointing%20device%20button%20(such%20as%20a%20mouse%27s%20primary%20mouse%20button)%20is%20both%20pressed%20and%20released%20while%20the%20pointer%20is%20located%20inside%20the%20element.)).

So basically:
1. The user presses down on the result item.
2. Browser sees the click on it and calls blur on the input element.
3. The event handler queues the removal of the box. (`destroy`)

Possible Scenario A:
4. The user leaves the mouse button and completes the click, browser fires the click event on the result.
5. The queued `destroy` call goes through and removes the box.

Possible Scenario B:
4. The queued `destroy` call goes through and removes the box.
5. The `click` event does not get called as the mouse press did not end on the same element (which no longer exists).
 
I suggest to not rely on the timeout and instead directly destroy the box after click and ignore the `blur` during this stage.

I can cook up a PR if needed.

**Comment by knadh** [comment]
Thanks for the inputs!

Tried #2 using the laptop trackpack and seems to fix it. @AndrewBastin this is what you're referring to as well, right?

**Comment by AndrewBastin** [comment]
@knadh somewhat on the right track, but it breaks another behaviour, see: https://github.com/knadh/autocomp.js/pull/2/files#r1310239741
